dataland-frontend/src/components/pages/MyDataRequestsOverview.vue

```diff
@@ -0,0 +1,513 @@
+      this.displayedData.sort((a, b) => this.customCompareForExtendedStoredDataRequests(a, b));
+      this.numberOfFilteredRequests = this.displayedData.length;
+      this.displayedData = this.displayedData.slice(
+        this.datasetsPerPage * this.currentPage,
+        this.datasetsPerPage * (1 + this.currentPage),
+      );
+      this.waitingForSearchResults = false;
+      window.scrollTo({
+        top: 0,
+        behavior: "smooth",
+      });
+    },
+    /**
+     * Compares the extended stored data requests (sort field, request status, last modified, company name)
+     * @param a ExtendedStoredDataRequest to sort
+     * @param b ExtendedStoredDataRequest to sort
+     * @returns result of the comparison
+     */
+    customCompareForExtendedStoredDataRequests(a: ExtendedStoredDataRequest, b: ExtendedStoredDataRequest) {
+      const aValue = a[this.sortField];
+      const bValue = b[this.sortField];
+
+      if (this.sortField != ("requestStatus" as keyof ExtendedStoredDataRequest) && aValue < bValue)
+        return -1 * this.sortOrder;
+      if (this.sortField != ("requestStatus" as keyof ExtendedStoredDataRequest) && aValue > bValue)
+        return this.sortOrder;
+
+      if (
+        (a.requestStatus == RequestStatus.Answered && b.requestStatus != RequestStatus.Answered) ||
+        (a.requestStatus == RequestStatus.Open && b.requestStatus == RequestStatus.Closed)
+      )
+        return -1;
+      if (
+        (b.requestStatus == RequestStatus.Answered && a.requestStatus != RequestStatus.Answered) ||
+        (b.requestStatus == RequestStatus.Open && a.requestStatus == RequestStatus.Closed)
+      )
+        return 1;
+
+      if (a.lastModifiedDate < b.lastModifiedDate) return 1;
+      if (a.lastModifiedDate > b.lastModifiedDate) return -1;
+
+      if (a.companyName < b.companyName) return -1;
+      else return 1;
+    },
+    /**
+     * Updates the data for the current page
+     * @param event event containing the new page
+     */
+    onPage(event: DataTablePageEvent) {
+      this.currentPage = event.page;
+      this.updateCurrentDisplayedData();
+    },
+    /**
+     * Converts dateString to date
+     * @param date unix time
+     * @returns string representing a date (DD.MM.YYYY)
+     */
+    convertDateStringToDate(date: number) {
+      const parsedDate = new Date(date);
+
+      const day = parsedDate.getDate();
+      const month = parsedDate.getMonth() + 1;
+      const year = parsedDate.getFullYear();
+
+      const paddedDay = day < 10 ? "0" + day : day;
+      const paddedMonth = month < 10 ? "0" + month : month;
+
+      return `${paddedDay}.${paddedMonth}.${year}`;
+    },
+    /**
+     * Converts dateString to time
+     * @param date unix time
+     * @returns string representing a time (HH:MM)
+     */
+    convertDateStringToTime(date: number) {
+      const dateString = convertUnixTimeInMsToDateString(date);
+      return dateString.split(",")[2].trim();
+    },
```

Comment:
Falls es diese Funktionen noch nicht in einem utils - skript gibt können die gerne in ein utils stkript rein. Könnte man realistisch auch auf anderen Seiten gut gebrauchen.

Reply:
sind jetzt bei convertUnixTimeInMsToDateString in DataFormatUtils

Reply:
reminder für mich: unit testing das?

Reply:
Da wir jetzt das Datum wie bei "MY DATASETS" anzeigen, existieren die Funktionen nicht mehr!